src/Components/Web.Extensions/src/InputFile/IFileListEntry.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    /// <summary>
+    /// Represents the data of a file selected from an <see cref="InputFile"/> component.
+    /// </summary>
+    public interface IFileListEntry
+    {
+        /// <summary>
+        /// Gets the name of the file.
+        /// </summary>
+        string? Name { get; }
+
+        /// <summary>
+        /// Gets the last modified date.
+        /// </summary>
+        DateTime? LastModified { get; }
+
+        /// <summary>
+        /// Gets the size of the file in bytes.
+        /// </summary>
+        long Size { get; }
+
+        /// <summary>
+        /// Gets the MIME type of the file.
+        /// </summary>
+        string? Type { get; }
+
+        /// <summary>
+        /// Gets the file's path relative to the base directory selected in a directory picker.
+        /// This is not supported on all browsers: see
+        /// https://developer.mozilla.org/en-US/docs/Web/API/File/webkitRelativePath.
+        /// </summary>
+        string? RelativePath { get; }
+
+        /// <summary>
+        /// Gets a stream of the file's data.
+        /// </summary>
+        Stream Data { get; }
```

Comment:
Where does this get set? Could we call this `OpenReadStream` to keep with `M.Extensions.FileProvider.IFile`? It's a bit weird to see a Stream as a readable property

Reply:
We were discussing a similar change [here](https://github.com/dotnet/aspnetcore/pull/24640#discussion_r468081456).